creator-node/src/fileManager.js

```diff
@@ -748,6 +753,175 @@ async function removeFile(storagePath) {
   }
 }
 
+async function deleteFileOrDir(pathToFileOrDir) {
+  // Base case - delete single file (not a directory)
+  if (!(await fs.lstat(pathToFileOrDir)).isDirectory()) {
+    await fs.unlink(pathToFileOrDir)
+    return
+  }
+
+  // Recursively remove all contents of directory
+  for (const file of await fs.readdir(pathToFileOrDir)) {
+    const childPath = path.join(pathToFileOrDir, file)
+    if ((await fs.lstat(childPath)).isDirectory()) {
+      await deleteFileOrDir(childPath)
+    } else {
+      await fs.unlink(childPath)
+    }
+  }
+
+  // Remove actual directory
+  await fs.rmdir(pathToFileOrDir)
+}
+
+async function batchDeleteFileOrDir(storagePaths, batchSize, logger) {
+  let numFilesDeleted = 0
+  const batches = chunk(storagePaths, batchSize)
+  const promiseResults = []
+  for (const batchOfStoragePaths of batches) {
+    const promiseResultsForBatch = await Promise.allSettled(
+      batchOfStoragePaths.map((storagePath) => deleteFileOrDir(storagePath))
+    )
+    promiseResults.push(...promiseResultsForBatch)
+  }
+
+  // Count number of files successfully deleted
+  for (const promiseResult of promiseResults) {
+    if (promiseResult.status === 'fulfilled') {
+      numFilesDeleted++
+    } else {
+      logger.error(`Could not delete file: ${promiseResult?.reason?.stack}`)
+    }
+  }
+  return numFilesDeleted
+}
+
+/**
+ * Adds path to redis set for every file of the given user.
+ * DOES NOT DELETE. Call deleteAllCNodeUserDataFromDisk() to delete the data that was added to redis.
+ * Uses pagination to avoid loading all files in memory for users with a lot of data.
+ * @param {string} walletPublicKey the wallet of the user to delete all data for
+ * @param {bunyan.Logger} logger the logger to print messages to
+ * @return number of file paths added to redis
+ */
+async function gatherCNodeUserDataToDelete(walletPublicKey, logger) {
```

Comment:
since this has sql statements we should put this in dbManager.js

Reply:
it's part of the same logic as the surrounding functions tho - wdyt about moving it all to diskManager? it fetches from SQL but it doesn't update db and its purpose is to delete from disk

Reply:
yes they would be great!